content/blog/prototyping/index.md

```diff
@@ -0,0 +1,767 @@
+- no compilation step 
+
+The goal is to get as close to that experience in Rust as possible while staying true to Rust's core principles.
+Let's make changes quick and painless and rapidly iterate on your design without painting ourselves into a corner.
+
+## Tips And Tricks For Prototyping In Rust
+
+### Start small
+
+There's some overlap between prototyping and "[easy Rust](https://www.youtube.com/watch?v=33FG6O3qejM)."
+
+Allow yourself to ignore some of the best practices for production code for a while.
+
+It's possible, but you need to switch off your inner critic who always wants to write perfect code from the beginning.
+Rust enables you to comfortably defer perfection.
+You can make the rough edges obvious so that you can sort them out later.
+Don't let perfect be the enemy of good.
+
+One of the biggest mistakes I observe is an engineer's perfectionist instinct to jump on minor details which don't have a broad enough impact to warrant the effort.
+It's better to have a working prototype with a few rough edges than a perfect implementation of a small part of the system.
+
+Remember: you are exploring!
+Use a coarse brush to paint the landscape first.
+Try to get into a flow state where you can quickly iterate.
+Don't get distracted by the details too early.
+During this phase, it's also fine to throw away a lot failed attempts. 
+
+### Use Simple Types
+
+Even while prototyping, the type system is not going away.
+There are ways to make this a blessing rather than a curse.
+
+Use simple types like `i32`, `String`, `Vec` in the beginning.
+We can always make things more complex later if we have to -- the reverse is much harder.
+
+Here's a quick reference for common prototype-to-production type transitions:
+
+| Prototype        | Production          | When to switch                                                                     |
+| ---------------- | ------------------- | ---------------------------------------------------------------------------------- |
+| `String`         | `&str`              | When you need to avoid allocations or store string data with a clear lifetime      |
+| `Vec<T>`         | `&[T]`              | When the owned vector becomes too expensive to clone or you can't afford the heap  |
+| `Box<T>`         | `&T` or `&mut T`    | When `Box` becomes a bottleneck or you don't want to deal with heap allocations    | 
+| `Rc<T>`          | `&T`                | When the reference counting overhead becomes too expensive or you need mutability  | 
+| `Arc<Mutex<T>>`  | `&mut T`            | When you can guarantee exclusive access and don't need thread safety               |
+
+These owned types sidestep most ownership and lifetime issues, but they do it by allocating memory on the heap - just like Python or JavaScript would.
+
+You can always refactor when you actually need the performance or tighter resource usage, but chances are you won't.
+
+### Make use of type inference
```

Comment:
```suggestion
### Make Use of Type Inference
```

(Or leave it as-is and change the other ones. But mixing the two styles at the same level of header is weird.)